File: cpp-alg/leecode/Q35_search_insert_position.cpp
#include <iostream>
#include <vector>

int searchInsert(std::vector<int> &nums, int target) {
  int n = nums.size();
  int left = 0, right = n - 1;
  while (left <= right) {
    int mid = left + (right - left) / 2;
    if (nums[mid] < target)
      left = mid + 1;
    else
      right = mid - 1;
  }
  return left;
}

int main() {
  std::vector<int> array = {1, 3, 5, 6};
  int target = 7;
  int ans = searchInsert(array, target);
  std::cout << ans << std::endl;
}

File: python-alg/leecode/q35_search_insert_position.py
def search_insert_position(arr, target):
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return left


my_list = [3, 9, 10, 27, 38, 43, 82]
print(search_insert_position(my_list, 12))
